tempodb/encoding/vparquet/block_traceql.go

```diff
@@ -0,0 +1,1018 @@
+	columnPathSpanAttrKey        = "rs.ils.Spans.Attrs.Key"
+	columnPathSpanAttrString     = "rs.ils.Spans.Attrs.Value"
+	columnPathSpanAttrInt        = "rs.ils.Spans.Attrs.ValueInt"
+	columnPathSpanAttrDouble     = "rs.ils.Spans.Attrs.ValueDouble"
+	columnPathSpanAttrBool       = "rs.ils.Spans.Attrs.ValueBool"
+	columnPathSpanHttpStatusCode = "rs.ils.Spans.HttpStatusCode"
+	columnPathSpanHttpMethod     = "rs.ils.Spans.HttpMethod"
+	columnPathSpanHttpUrl        = "rs.ils.Spans.HttpUrl"
+)
+
+var intrinsicDefaultScope = map[traceql.Intrinsic]traceql.AttributeScope{
+	traceql.IntrinsicName:     traceql.AttributeScopeSpan,
+	traceql.IntrinsicDuration: traceql.AttributeScopeSpan,
+}
+
+// Lookup table of all well-known attributes with dedicated columns
+var wellKnownColumnLookups = map[string]struct {
+	columnPath string                 // path.to.column
+	level      traceql.AttributeScope // span or resource level
+	typ        traceql.StaticType     // Data type
+}{
+	// Resource-level columns
+	LabelServiceName:      {columnPathResourceServiceName, traceql.AttributeScopeResource, traceql.TypeString},
+	LabelCluster:          {columnPathResourceCluster, traceql.AttributeScopeResource, traceql.TypeString},
+	LabelNamespace:        {columnPathResourceNamespace, traceql.AttributeScopeResource, traceql.TypeString},
+	LabelPod:              {columnPathResourcePod, traceql.AttributeScopeResource, traceql.TypeString},
+	LabelContainer:        {columnPathResourceContainer, traceql.AttributeScopeResource, traceql.TypeString},
+	LabelK8sClusterName:   {columnPathResourceK8sClusterName, traceql.AttributeScopeResource, traceql.TypeString},
+	LabelK8sNamespaceName: {columnPathResourceK8sNamespaceName, traceql.AttributeScopeResource, traceql.TypeString},
+	LabelK8sPodName:       {columnPathResourceK8sPodName, traceql.AttributeScopeResource, traceql.TypeString},
+	LabelK8sContainerName: {columnPathResourceK8sContainerName, traceql.AttributeScopeResource, traceql.TypeString},
+
+	// Span-level columns
+	LabelHTTPStatusCode: {columnPathSpanHttpStatusCode, traceql.AttributeScopeSpan, traceql.TypeInt},
+	LabelHTTPMethod:     {columnPathSpanHttpMethod, traceql.AttributeScopeSpan, traceql.TypeString},
+	LabelHTTPUrl:        {columnPathSpanHttpUrl, traceql.AttributeScopeSpan, traceql.TypeString},
+}
+
+// Fetch spansets from the block for the given TraceQL FetchSpansRequest. The request is checked for
+// internal consistencies:  operand count matches the operation, all operands in each condition are identical
+// types, and the operand type is compatible with the operation.
+func (b *backendBlock) Fetch(ctx context.Context, req traceql.FetchSpansRequest) (traceql.FetchSpansResponse, error) {
+
+	err := checkConditions(req.Conditions)
+	if err != nil {
+		return traceql.FetchSpansResponse{}, errors.Wrap(err, "conditions invalid")
+	}
+
+	rr := NewBackendReaderAt(ctx, b.r, DataFileName, b.meta.BlockID, b.meta.TenantID)
```

Comment:
use `openForSearch()`? We've consolidate the reader building logic nicely there.